### PR TITLE
Condor tail addition

### PR DESCRIPTION
**Pull request by gurudattapatil**
i have checked the working of this on simple condor jobs and seems to be working fine for it.

### CODE REVIEW

File: ganga/GangaCore/Lib/Condor/Condor.py [hunk 1]
```diff
@@ -797,4 +797,38 @@ def timedetails(self):
     updateMonitoringInformation = \
         staticmethod(updateMonitoringInformation)
 
+    def peek(self, filename=None, command=None):
+        job = self.getJobObject()
+
+        if not self.id:
+            logger.warning("Job %s not running" % job.id)
+            return False
```
Comment (egede) [inline ganga/GangaCore/Lib/Condor/Condor.py]: We should use the job status to decide on this. But is the check actually required? Don't we only end up here if the job is in the `running` state?
```suggestion
        if job.status!='running':
            logger.warning("Job %s not running" % job.id)
            return False
```

File: ganga/GangaCore/Lib/Condor/Condor.py [hunk 2]
```diff
@@ -797,4 +797,38 @@ def timedetails(self):
...
+
+        idElementList = job.backend.id.split("#")
+        if 3 == len(idElementList):
+            if idElementList[1].find(".") != -1:
+                peekCommand = "condor_tail -f -name %s %s" % \
+                              (idElementList[0], idElementList[1])
```
Comment (egede) [inline ganga/GangaCore/Lib/Condor/Condor.py]: I suggest to use `f` strings instead as they are more readable. Same change in several other places in code.

```suggestion
                peekCommand = f"condor_tail -f -name {idElementList[0]} {idElementList[1]}"
```

File: ganga/GangaCore/Lib/Condor/Condor.py [hunk 3]
```diff
@@ -797,4 +797,38 @@ def timedetails(self):
     updateMonitoringInformation = \
         staticmethod(updateMonitoringInformation)
 
+    def peek(self, filename=None, command=None):
+        job = self.getJobObject()
+
```
Comment (egede) [inline ganga/GangaCore/Lib/Condor/Condor.py:802]: What happens if you call `peek` on a master job (where there is no actual job running, so nothing to `peek`). You need to check if your `job` object has subjobs, and if yes, return `False`.

**Review by egede — CHANGES_REQUESTED** [review]

File: ganga/GangaCore/Lib/Condor/Condor.py (hunk 1 above)
Comment (gurudattapatil) [reply ganga/GangaCore/Lib/Condor/Condor.py]: I will remove this as it may not be required.

File: ganga/GangaCore/Lib/Condor/Condor.py (hunk 2 above)
Comment (gurudattapatil) [reply ganga/GangaCore/Lib/Condor/Condor.py]: Done

File: ganga/GangaCore/Lib/Condor/Condor.py (hunk 3 above)
Comment (gurudattapatil) [reply ganga/GangaCore/Lib/Condor/Condor.py:802]: Done

File: ganga/GangaCore/Lib/Condor/Condor.py [hunk 4]
```diff
@@ -797,4 +797,38 @@ def timedetails(self):
...
+            logger.warning("Tried command: '%s'" % peekCommand)
+            logger.warning("Command output: '%s'" % output)
+        else:
+            logger.info(output)
+
+        peekStatus = True
+        if job.subjobs:
+            peekStatus= False
+            logger.error("Master Job does not have a peek status.")
+
```
Comment (egede) [inline ganga/GangaCore/Lib/Condor/Condor.py]: Should this not be up at the top at line 803 instead? Code also seems a bit inconsistent in if for a failure returns

File: ganga/GangaCore/Lib/Condor/Condor.py [hunk 5]
```diff
@@ -797,4 +797,38 @@ def timedetails(self):
...
+        status, output = subprocess.getstatusoutput(peekCommand)
+
+
+        if (status != 0):
+            logger.warning \
+                ("Return code '%s' peek job '%s' - Condor id '%s'" %
+                 (str(status), job.id, job.backend.id))
+            logger.warning("Tried command: '%s'" % peekCommand)
+            logger.warning("Command output: '%s'" % output)
```
Comment (egede) [inline ganga/GangaCore/Lib/Condor/Condor.py:823]: This should cause `peekStatus = False`

**Review by egede — CHANGES_REQUESTED** [review]

**Pull request merged**